### PR TITLE
parametrization: prevent dumping parametrized stages

### PR DESCRIPTION
**Pull request by skshetry**
This is a bit hackish, but does the job done.
It feels like we need some unified way to be able to load
and dump, but not sure how. Till then, this data class
will be helpful in communicating between those.

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/stage/__init__.py [hunk 1]
```diff
@@ -131,6 +139,7 @@ def __init__(
         self._stage_text = stage_text
         self._dvcfile = dvcfile
         self.desc = desc
+        self.meta = Meta()
```
Comment (efiop) [inline dvc/stage/__init__.py:142]: This is not the same as `meta` field, right? If so, could we change the name? It is really confusing otherwise.

Comment (skshetry) [reply dvc/stage/__init__.py:142]: No, we don't keep `meta` around. It's only on `_stage_text`. Will change it to `RawData`/`raw_data`.

File: dvc/stage/loader.py [hunk 2]
```diff
@@ -106,13 +107,23 @@ def __getitem__(self, name):
...
+        stage = self.load_stage(
             self.dvcfile,
             name,
             self.resolved_data[name],
             self.lockfile_data.get(name, {}),
         )
 
+        group, *keys = name.rsplit(JOIN, maxsplit=1)
+        if group and keys and name in self.stages_data:
+            stage.raw_data.generated_from = group
```
Comment (pared) [inline dvc/stage/loader.py:119]: It seems to me that in the test case `build@us` `generated_from` is not set - shouldn't it be set to `build`?

**Review by pared — COMMENTED** [review]
One question

File: dvc/stage/loader.py (hunk 2 above)
Comment (skshetry) [reply dvc/stage/loader.py:119]: Silly mistake. It should have been
```suggestion
        if group and keys and name not in self.stages_data:
            stage.raw_data.generated_from = group
```

**Review by pared — APPROVED** [review]

**Review by efiop — APPROVED** [review]

**Pull request merged**